addons/web/static/src/core/signature/name_and_signature.js

```diff
@@ -123,6 +123,16 @@ export class NameAndSignature extends Component {
         this.props.signature.isSignatureEmpty = this.isSignatureEmpty;
     }
 
+    /**
+    * Loads a signature image from a base64 dataURL and updates the empty state.
+    */
+    fromDataURL(dataURL, options = {}) {
+        this.signaturePad.fromDataURL(dataURL, options).then(() => {
```

Comment:
Also, this promise is neither awaited nor returned. So, when calling the `NameAndSignature#fromDataURL()` function is called, we have no way to properly wait for it to be finished.